File: LeetCode Problems/1101 to 1200/1137. N-th Tribonacci Number/1137. N-th Tribonacci Number.kt
class Solution {
    fun tribonacci(n: Int): Int {
        if(n == 0) return 0;
        else if (n <= 2) return 1;
        var a = 0; var b = 1; var c = 1; var ans = 0;  
        for (i in 3..n) {
            ans = a + b + c;
            a = b; b = c; c = ans;
        }
        return ans;
    }
}

File: LeetCode Problems/1101 to 1200/1137. N-th Tribonacci Number/1137. N-th Tribonacci Number.py
class Solution:
    def tribonacci(self, n: int) -> int:
        if n == 0: return 0
        elif n <= 2: return 1
        a = 0;b = 1;c = 1;ans = 0
        for i in range (3, n+1):
            ans = a + b + c
            a = b;b = c;c = ans
        return ans